src/algorithms/uncategorized/huffman-coding/__test__/huffmanCoding.test.js

```diff
@@ -0,0 +1,133 @@
+var Heap = function(fn) {
+    this.fn = fn || function(e) {
+      return e;
+    };
+    this.items = [];
+};
+  
+Heap.prototype = {
+    swap: function(i, j) {
+        this.items[i] = [
+        this.items[j],
+        this.items[j] = this.items[i]
+        ][0];
+    },
+    bubble: function(index) {
+        var parent = ~~((index - 1) / 2);
+        if (this.item(parent) < this.item(index)) {
+        this.swap(index, parent);
+        this.bubble(parent);
+        }
+    },
+    item: function(index) {
+        return this.fn(this.items[index]);
+    },
+    pop: function() {
+        return this.items.pop();
+    },
+    sift: function(index, end) {
+        var child = index * 2 + 1;
+        if (child < end) {
+        if (child + 1 < end && this.item(child + 1) > this.item(child)) {
+            child++;
+        }
+        if (this.item(index) < this.item(child)) {
+            this.swap(index, child);
+            return this.sift(child, end);
+        }
+        }
+    },
+    push: function() {
+        var lastIndex = this.items.length;
+        for (var i = 0; i < arguments.length; i++) {
+        this.items.push(arguments[i]);
+        this.bubble(lastIndex++);
+        }
+    },
+    get length() {
+        return this.items.length;
+    }
+};
+  
+var Huffman = {
+    // encode function
+    encode: function(data) {
+        var prob = {};
+        var tree = new Heap(function(e) {
+        return e[0];
+        });
```

Comment:
Something is wrong formatting here :thinking:

Reply:
okay! I think it is better to use the arrow function.
Thank you! 😄